### DISCUSSION TITLE
flashing Tasmota on Shelly Pro 1PM (DIN rail), how to?

### DISCUSSION THREAD

**Discussion by Eisbahn** [arendst/Tasmota#17669] [Q&A] [answered]
Dear all,

just got a new Shelly Pro 1PM (DIN rail mount) which should be flashed with Tasmota. At the right side of the housing I could see some small openings (7 in total) and it looks like some debug pins are behind it. However: I've no clue about the pinout.
Has anybody flashed Tasmota to a Shelly Pro 1PM and could share some advice how to do it?
For the "non-Pro" there are some hints available, but for the Pro/DIN rail mount ones I couldn't find anything.
![20230110_123911](https://user-images.githubusercontent.com/336105/211543050-57a694c4-2c4a-4158-a00b-2f380611dae6.jpg)

Thanks a lot for help

**Comment by Noschvie** [comment, marked as answer]
[Pinout](https://user-images.githubusercontent.com/38843794/156607973-259d2b27-976c-4b71-9198-ec737f6e95fb.png)

**Comment by Eisbahn** [comment]
Thanks a lot for your quick answer, made my day and worked perfectly.
Thanks!

**Comment by Eisbahn** [comment]
ok, flashing worked, but config is horrible wrong/not working:
- the Auto-config with Shelly Plus 1PM does not work at all (no toggeling of relais, no voltage, no current, no power info, no temp...). Ok, very likely Plus isn't the same as Pro ;-)
- But also this one does not work nicely: https://templates.blakadder.com/shelly_pro_1pm.html
toggeling the relais and temp info is present and seeems to be perfect, frequency info is there but horrible wrong (something like 4Hz on an expected 50Hz german power line...) and voltage, current drawn and power is missing completely (always zero, regardless if a load is driven or not).

Any advice how to set it up properly? Sadly also some/most of the "numbers" in the config are not documented, therefore I don't know what to do...

**Reply by barbudor** [reply]
Sorry I missed you listed it

You have this guide which may help : https://tasmota.github.io/docs/Configuration-Procedure-for-New-Devices/

But personally I would go for opening and looking for what's inside

**Reply by sfromis** [reply]
(You can see the meaning of the numbers on the template page, the table to the right, and in "Configure Tempate")

**Reply by Eisbahn** [reply]
> (You can see the meaning of the numbers on the template page, the table to the right, and in "Configure Tempate")

you mean this one: https://tasmota.github.io/docs/Components/#assigning-components
Even the first number/component (9568) isn't listed, neither for ESP32, nor on ESP8266 chapter

**Reply by barbudor** [reply]
he was meaning this one: 

![image](https://user-images.githubusercontent.com/2996042/211648262-2644340d-b74f-4577-b4a0-9ae36f653e3e.png)

**Reply by sfromis** [reply]
Yes, I meant the specific places, not that other page which does not seem to be quite updated. Anyway, here's the template decoded, including numbers. This matches what you can see the two places I mentioned.
```
Gpio.00     IO 9568 ADE7953 CS1
Gpio.01     TX    1 User
Gpio.02     IO 9472 ADE7953 RST
Gpio.03     RX    1 User
Gpio.04     IO  768 SPI CS1
Gpio.05     IO    0 None
Gpio.09     FL    0 None
Gpio.10     FL    0 None
Gpio.12     IO  672 SPI MISO1
Gpio.13     IO  704 SPI MOSI1
Gpio.14     IO  736 SPI CLK1
Gpio.15     IO    0 None
Gpio.16     IO    0 None
Gpio.17     IO    0 None
Gpio.18     IO 5600 ETH MDIO
Gpio.19     IO 6214 Option A7
Gpio.20     IO    0 None
Gpio.21     IO    0 None
Gpio.22     IO    0 None
Gpio.23     IO 5568 ETH MDC
Gpio.24     IO    0 None
Gpio.25     IO    0 None
Gpio.26     IO    0 None
Gpio.27     IO    0 None
Gpio.06     FL    0 None
Gpio.07     FL    0 None
Gpio.08     FL    0 None
Gpio.11     FL    0 None
Gpio.32     AO 3459 ADE7953 IRQ4
Gpio.33     AO    0 None
Gpio.34     IA    0 None
Gpio.35     IA   32 Button1
Gpio.36     IA 4736 ADC Temp1
Gpio.37     IA    0 None
Gpio.38     IA  160 Switch1
Gpio.39     IA    0 None
```

**Reply by Eisbahn** [reply]
Hi,

this is exactly what I'm using and what is not working at all (beside the sensless temp info and relaif toggeling). In this case I could also use a quite cheaper shelly with just a relais...
Are there any commands which need to be made, some config which needs to be done during start up, some adaption of paraemters e.g. with an old bulb of 50W or similar?

**Reply by barbudor** [reply]
There is a possibility that you have a new hardware revision with some changes
Do you have the opportunity to open it and check if you can confirm that the ADE7953 is still there ?
What temperature are you refering too ? If it s the ESP32 internal one, it is known to be useless

**Reply by arendst** [reply]
There are indeed two versions of the devices. See Shelly knowledgebase. Tasmota should support the latest version.

Info from the knowledgebase:

```
Shelly Pro 1PM v.1 is a modification of Shelly Pro 1PM (SPSW-001PE16EU)

Shelly Pro 1PM is now obsolete and out of sale.
[Download Shelly Pro 1PM multi-language printed user manual](https://shelly.cloud/documents/user_guide/Shelly_PRO_1PM_multilanguage_v04_web.pdf).

Differences with Shelly Pro 1PM are marked by the ≠ symbol in the text below.

Main changes:

Power supply: no more 12 VDC option.

Connectors: 3-terminal connectors are replaced by 2-terminal ones.

PCBs: relay 2-layer PCBs are replaced by 4-layer ones for better thermal performance.

LAN: improved high voltage electrical distances.

Plastics shell: improved dielectric performance.
```

A quick external identification is the change from a three terminal connector to a two terminal connector.

**Reply by Eisbahn** [reply]
> There are indeed two versions of the devices. See Shelly knowledgebase. Tasmota should support the latest version.
> 
> A quick external identification is the change from a three terminal connector to a two terminal connector.

I do have the newer v1 (just ordered/received the Shelly) with just 2 terminal connector and no 12VDC support (which is fine for me)

**Reply by arendst** [reply]
OK. So in that case open it up and let's see if it's much different from this (Shelly Pro 2PM)

![IMG_20230111_105507899_HDR](https://user-images.githubusercontent.com/11044339/211775355-f4cda6d7-9de6-47e7-a1f9-ad355aae211b.jpg)

**Reply by arendst** [reply]
I just ordered a Pro 1PM too. Let's see if it differs too much from the Pro 2PM....

**Comment by Eisbahn** [comment]
That looks quite common. Is there anything else I need to do after flashing the device?

![20230111_164736](https://user-images.githubusercontent.com/336105/211854029-58c43708-82d1-4298-8e32-a653111dddd1.jpg)
![20230111_164837](https://user-images.githubusercontent.com/336105/211854079-1b027fe6-59b1-4f9d-b683-8459fdb0fc91.jpg)
![20230111_164908](https://user-images.githubusercontent.com/336105/211854085-89ecfc7b-6a6b-4ed5-b6b8-c0581ac57ab1.jpg)
![20230111_164922](https://user-images.githubusercontent.com/336105/211854110-5f85a339-d28d-45c7-a5c1-7b5f318e5269.jpg)
![20230111_164934](https://user-images.githubusercontent.com/336105/211854124-7267d539-5773-4554-9cdd-55513b7b4946.jpg)
![20230111_165005](https://user-images.githubusercontent.com/336105/211854134-e41614af-f1f4-4d53-9e63-778365c48e90.jpg)
![20230111_165032](https://user-images.githubusercontent.com/336105/211854153-7c72b45c-5d8c-4bff-8f1a-d43e7ddcfda8.jpg)

**Reply by arendst** [reply]
After flashing just enable the correct template and load Auto config for Shelly Pro 1PM.

I see your hardware is two months older than mine. I have an open Shelly Pro 2 and a Shelly Pro 2PM. Both mine and yours have a different main board. The difference is around the reset logic. In your case there is a (yellow) subshield mounted. My 2 misses a transistor controlling the reset signal.

So something maybe fishy around your reset circuit.

BTW did you ever connect it to AC? If not you won't see any valid energy monitoring information. This means connecting AC to one side (L and N) AND connect L to the I connector.

**Reply by Eisbahn** [reply]
In which release do I find the correct template? Took the nightly tasmota32.bin and there is only the Shelly Plus 1PM inside, there is no Pro (see my second posting). Reset works fine, at least from the HMI/webinterface the Shelly makes after reconfiguration a restart.
And yes, I fully connected the Shelly, only the two switch inputs were never connected. If I do so, I do also get frequnecy info (which is horrible wrong), but no power, voltage or current (always zero)

**Reply by sfromis** [reply]
The templates are not in the builds, look for your specific device in https://templates.blakadder.com/
Some devices can also be configured via auto-configuration, also providing the template.

**Reply by Noschvie** [reply]
[SetOption21](https://tasmota.github.io/docs/Commands/#setoption21) - Energy monitoring when power is off

**Reply by arendst** [reply]
What you see as energy monitoring is typical the result of missing AC power at the monitoring device. if only DC is present you see a low frequency and no voltage/current/power.

Did the device work with the shelly firmware?

**Reply by arendst** [reply]
FWIW I received my shelly Pro 1PM today and at first looks the PCA's look the same as yours. Currently backing up the 8M Shelly firmware before flashing Tasmota.

Just a note, how did you connect AC to the blue bottom connector. It should be like this: 

![IMG_20230112_155109376](https://user-images.githubusercontent.com/11044339/212098965-77315321-627a-4838-9b9b-ac5923343fee.jpg)

where brown (L) on the left and blue (N) on the right.

**Reply by arendst** [reply]
Installed tasmota with no issue at all. Except for wrong connections I have no more clue what could be your issue.

**Comment by Eisbahn** [comment]
Sorry for delay, have some busy working hours.
With the Shelly FW everything worked, sadly I haven't backed up the original FW. With Tasmota (and same wiring) I do get still the same odd results. Of course 230V AC was disconnected while flashing ;-)
Will tomorrow double check it and e.g. swap N and L on inputs and will measure it also again. The Shelly device  itself is working and also showing the power LED and also the load is running. All on 230V AC (german power line), I don't have any DC (as e.g. 12V DC isn't supported by mine at all)

**Comment by Eisbahn** [comment]
Thanks to Am**on returning the device was possible. The new/exchanged one is working like a charm exaclty with same wiring. Most likely it had a defect on arrival.
Thanks for your support and help on this one.

**Comment by RolandJonkers** [comment]
Not sure if this thread is still open but I need some advice.
I flashed the Shelly pro 1PM by means of the Tasmota online flash tool. All went successfully according to the SW.
But the wifi LED is not turning on and the AP is not visible so I can not connect.
What am I doing wrong? Thanks for the support

**Reply by Noschvie** [reply]
How is your device connected after the flash process? By 220V ?

**Reply by RolandJonkers** [reply]
Yes correct. Unplugged the converter and connected to 220V

**Reply by sfromis** [reply]
As you gave no operational and exact details of your steps (like specific tool and file used), hard to guess what you may have been doing wrong. Seeing a message saying that "something" was successful does not imply that your steps were correct, or if it was an actual "success" at all.

**Reply by RolandJonkers** [reply]
I am learning on the job....
I flashed a couple of Shelly 1 in the past. 

I bought a new CH340G converter because I wasn't sure the old one was still okay.
First I flashed without selecting any .bin file assuming it would use the right version.
Then I selected a tasmota32.bin myself. And I tried the OTA version.
Do I have to use the OTA version? Or the factory version? I am a bit confused now.
Maybe it is just a matter of picking the correct .bin file.
Hope you are able to help 

Oh and I am using the Tasmota Web installer

**Reply by sfromis** [reply]
Well, avoiding wrong directions like flashing `tasmota32.bin` is a good reason to use the recommended tool https://tasmota.github.io/install/ and just not making any selection, as it should detect what binary is needed. In case of the original ESP32, that is `tasmota32.factory.bin`

**Reply by RolandJonkers** [reply]
Thanks for the support. Finally i got it downloaded and working.
All works except for the power readout. Always reading zero. Voltage and frequency works, but no power. Tried old FW 13, release 14.5 ruth and now latest dev build 14.5.0.3.   Default tasmota32.bin build. Still no succes. Tried to calibrate the power but no effect. Re-applied the configuration. Still no power. Looks like the ADE9753 not working? I do see a current readout in the console but reading is   _Channel1 ACCMODE 0x000000, VRMS 6345183, Period 4473, IRMS 1773, WATT 0, VA 29, VAR 0_   thus 1.77mA. And it does not make a difference with load on or off. Is the config changed? My module defective or am I doing something wrong?
Thanks for any advice!

![image](https://github.com/user-attachments/assets/cf5daa42-2d5d-4e93-b1ab-3375848cd89e)

**Reply by RolandJonkers** [reply]
update: the issue turned out to be a faulty module. After replacement everything works as expected.

**Comment by Letmeload** [comment]
Hello, I have the same problem as [RolandJonkers](https://github.com/arendst/Tasmota/discussions/17669#discussioncomment-12725927).
I can't get any power and current data.
Firstly, I would like to point out that I have a Shelly Pro 1PM V1 and not a V0.
https://kb.shelly.cloud/knowledge-base/shelly-pro-1pm-v1

Flasehd the PRO1PM V1 OTA with this instruction:
https://github.com/tasmota/mgos32-to-tasmota32
Previously, the Shelly firmware was on v1.55

The device hangs in safeboot mode. After I flashed the tasmota32.bin from the github page linked above I got the device running.
You cant find it... download it here:
[tasmota32.zip](https://github.com/user-attachments/files/20128949/tasmota32.zip)

Then I updated it to the latest Tasmota 14.6.0 firmware and used the [template from blakadder](https://templates.blakadder.com/shelly_pro_1pm.html).
`{"NAME":"Shelly Pro 1PM","GPIO":[9568,1,9472,1,768,0,0,0,672,704,736,0,0,0,5600,6214,0,0,0,5568,0,0,0,0,0,0,0,0,3459,0,0,32,4736,0,160,0],"FLAG":0,"BASE":1,"CMND":"AdcParam1 2,10000,10000,3350"}`
Perhaps this would have to be rewritten for the V1 so that the energy monitor works?
setoption21 is on does his job.

Here more Data from the Shelly PRO 1PM v.1
![Shelly 0](https://github.com/user-attachments/assets/2ff3c27d-bfa0-40cb-9f5d-9864f1cee630)
![Shelly 1](https://github.com/user-attachments/assets/b63db7c6-d42f-4b4e-8221-1721e6a034c5)
![Shelly 2](https://github.com/user-attachments/assets/e139fcb6-f128-4cf9-af3e-87181a84586c)
![Shelly 3](https://github.com/user-attachments/assets/c344ef6d-90a7-43ff-a8b4-dad8fc5427f6)

**Reply by Letmeload** [reply]
It was actually also a defective newly purchased Shelly PRO 1PM. It's a pity that the flashing voided the warranty.
